java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/ConnectionTlsTest.java

```diff
@@ -43,82 +40,59 @@
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.ClassRule;
 import org.junit.Test;
 
-import com.google.common.base.Strings;
-
 /**
  * Tests encrypted connections.
- * TODO Update to use {@link FlightServerTestRule} instead of {@link FlightTestUtils}
  */
 public class ConnectionTlsTest {
+
+  @ClassRule
+  public static final FlightServerTestRule FLIGHT_SERVER_TEST_RULE;
+  private static final MockFlightSqlProducer PRODUCER = new MockFlightSqlProducer();
+
+  static {
+    FlightServerTestRule.CertKeyPair certKey = null;
+    try {
+      certKey = exampleTlsCerts().get(0);
+    } catch (URISyntaxException e) {
+      System.out.println("The syntax of Certificate is invalid: " + e.getMessage());
+    }
+
+    UserPasswordAuthentication authentication =
+        new UserPasswordAuthentication.Builder().user("user1", "pass1").user("user2", "pass2")
+            .build();
+
+    FLIGHT_SERVER_TEST_RULE = new FlightServerTestRule.Builder()
+        .host("localhost")
+        .randomPort()
+        .authentication(authentication)
+        .useTls(certKey.cert, certKey.key)
+        .producer(PRODUCER)
+        .build();
+  }
+
   private final String keyStorePath = this.getClass().getResource("/keys/keyStore.jks")
       .getPath();
   private final String noCertificateKeyStorePath =
       this.getClass().getResource("/keys/noCertificate.jks")
           .getPath();
   private final String keyStorePass = "flight";
-  private FlightServer tlsServer;
-  private String serverUrl;
   private BufferAllocator allocator;
-  private FlightTestUtils flightTestUtils;
+  private ArrowFlightJdbcConnectionPoolDataSource dataSource;
 
   @Before
   public void setUp() throws Exception {
-    flightTestUtils = new FlightTestUtils("localhost", "flight1", "woho1",
-        "invalid", "wrong");
-
     allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
It seems that you are not using this allocator anymore. Please check it.

Reply:
allocator is in use.